body-pix/src/multi_person/person_masks_for_poses.ts

```diff
@@ -0,0 +1,314 @@
+import * as tf from '@tensorflow/tfjs-core';
+
+import {NUM_KEYPOINTS} from '../keypoints';
+import {Pose} from '../types';
+
+function getScale(
+    [height, width]: [number, number],
+    [inputResolutionY, inputResolutionX]: [number, number],
+    [[padT, padB], [padL, padR]]: [[number, number], [number, number]]):
+    [number, number] {
+  const scaleY = inputResolutionY / (padT + padB + height);
+  const scaleX = inputResolutionX / (padL + padR + width);
+  return [scaleX, scaleY];
+}
+
+export function personMasksForPosesGPU(
```

Comment:
Adds a quick test to make sure that CPU and GPU decoding can produce the same results.

Reply:
The testing runs in node.js on the cpu, so as of now it's challenging to test the comparison of both.

Also in general it would be hard to write a test like this for a few reasons:
* It is more of an integration test than a unit test as it is testing many layers at once.
* There is quite a bit of data setup , as well as verification setup.  
* There could be slight precision differences within the was gpu does float vs cpu.  Even if the code was the same, these slight differences could cause the tests to fail.

A better thing would be to write unit tests around the smaller pieces of functionality within those methods.

That being said - there weren't any tests on these methods to begin with, and I would't really have the time to do this, as it is pretty involved with all of the data setup and verification at the end.

Reply:
I know it's kinda hacky, but the way to verify it (without the automated tests) is to set `backend=cpu` in the querystring of the demo